core-bundle/src/Resources/contao/templates/backend/be_two_factor.html5

```diff
@@ -43,6 +43,36 @@
         </div>
       </div>
     </form>
+    <div class="tl_recovery_codes">
+      <h2 class="sub_headline"><?= $this->backupCodesLabel ?></h2>
+      <p><?= $this->backupCodesExplain ?></p>
+      <?php if (!$this->showBackupCodes): ?>
+        <form action="<?= $this->action ?>" class="tl_two_factor_form" method="post">
+          <input type="hidden" name="FORM_SUBMIT" value="tl_two_factor_show_backup_codes">
+          <input type="hidden" name="REQUEST_TOKEN" value="<?= REQUEST_TOKEN ?>">
+          <div class="submit_container cf">
+            <button type="submit" class="tl_submit"><?= empty($this->backupCodes) ? $this->backupCodesGenerate : $this->backupCodesShow ?></button>
+          </div>
+        </form>
+      <?php else: ?>
+        <div class="tl_message">
+          <p class="tl_info"><?= $this->backupCodesInfo ?></p>
+        </div>
+        <ul class="recovery-codes">
+          <?php foreach ($this->backupCodes as $backupCode): ?>
+            <li><code><?= $backupCode ?></code></li>
+          <?php endforeach; ?>
+        </ul>
+        <form action="<?= $this->action ?>" class="tl_two_factor_form" method="post">
+          <input type="hidden" name="FORM_SUBMIT" value="tl_two_factor_generate_backup_codes">
+          <input type="hidden" name="REQUEST_TOKEN" value="<?= REQUEST_TOKEN ?>">
+          <div class="submit_container cf">
+            <button type="submit" class="tl_submit"><?= $this->backupCodesRegenerate ?></button>
```

Comment:
According to our discussion, we should use $this->trans() for labels in the template

Reply:
Changed in a22ba09d3ac32e77bc78600b373a50e8bb283dc3.